quinn-h3/src/client.rs

```diff
@@ -173,10 +173,30 @@ impl SendRequest {
             recv: None,
         }
     }
+
+    fn build_response(&mut self, header: Header) -> Result<Response<RecvBody>, Error> {
+        let (status, headers) = header.into_response_parts()?;
+        let response = Response::builder()
+            .status(status)
+            .version(http::version::Version::HTTP_3)
+            .body(RecvBody::new(
+                try_take(&mut self.recv, "recv is none")?,
+                self.conn.clone(),
+                try_take(&mut self.stream_id, "stream is none")?,
+            ));
+
+        match response {
+            Err(e) => Err(Error::Peer(format!("invalid response: {:?}", e))),
```

Comment:
I'm still not clear on when this can possibly occur. `status` is already a `StatusCode` so parsing that won't fail, setting the HTTP version and body should be infallible, and the headers still aren't being checked.

Reply:
All header fields are checked: they are stored in `http` crate types at the time of decoding, every headers pass through [this validation](https://github.com/djc/quinn/blob/h3-streamed-body-server/quinn-h3/src/proto/headers.rs#L168), `http::HeaderName` and `http::HeaderValue` being the implementation I rely on for non-special fields. 

As this is the case for all other members of `Header`, I guess `.unwrap()`ing the `Request` and `Response` builders is not dangerous at all (only if the `quinn_h3::proto::headers` module do not change).

I just amended my last commit.

Reply:
`quinn_h3::proto::header::Field` has some minor problems tha I'mgoing to fix by the way.

Reply:
To be clear, by "not being checked" I meant by the calls made in this function, i.e. anything that could possibly have caused that error state to arise. Seems like that's been addressed.